lib/web/apiserver.go

```diff
@@ -1090,7 +1090,7 @@ func (h *Handler) createSession(w http.ResponseWriter, r *http.Request, p httpro
 		return nil, trace.AccessDenied("unknown second factor type: %q", cap.GetSecondFactor())
 	}
 	if err != nil {
-		return nil, trace.AccessDenied("bad auth credentials")
+		return nil, trace.AccessDenied("bad auth credentials: %v", err)
```

Comment:
Can the error here contain any sensitive information and is this error returned to the client? I'm inclined to log it instead of returning it in an error. That we we can be sure we're not accidentally returning sensitive information to the client.

Reply:
Good point. I changed this because debugging why login failed was impossible otherwise.
Changed to logging the error, it'll work for debugging too.